pkg/jwt/parser.go

```diff
@@ -0,0 +1,29 @@
+package jwt
+
+import (
+	"github.com/go-jose/go-jose/v4"
+	"github.com/go-jose/go-jose/v4/jwt"
+)
+
+// allAlgs contains all JOSE asymmetric signing algorithm values as defined by RFC 7518
+//
+// see: https://tools.ietf.org/html/rfc7518#section-3.1
+var allowedSignatureAlgorithms = []jose.SignatureAlgorithm{
```

Comment:
Is this essentially what v3 did?

Reply:
- https://github.com/go-jose/go-jose/issues/64#issuecomment-1809252335
- https://github.com/go-jose/go-jose/pull/74#issuecomment-2155241502

Before we can use any algorithm (I'm not sure if it was checked during parsing) now it's validated to be from the list.